nsv13/code/modules/jobs/job_types/marine/midshipman.dm

```diff
@@ -68,3 +68,55 @@ Marine & all their unique stuff!
 	name = "squad medic uniform"
 	desc = "A cheaply made and uncomfortable uniform worn by squad medics. It has a conspicuous blue cross on the back. Shooting its bearer may constitute a war crime."
 	icon_state = "marine_medic"
+
+/datum/job/assistant/after_spawn(mob/living/carbon/human/H, mob/M)
+	. = ..()
+	// Assign department
+	var/department
+	if(M && M.client && M.client.prefs && M.client.prefs.active_character)
+		department = M.client.prefs.active_character.preferred_security_department
+		if(department == "None")
+			return
+		else if(!(department in GLOB.available_depts))
+			department = pick(GLOB.available_depts)
```

Comment:
```suggestion
	var/department = M?.client?.prefs?.active_character?.preferred_security_department
	if(department == "None")
		to_chat(M, "<b>You have not been assigned to any department. Help in any way you can!</b>")
		return
	else if(!(department in GLOB.available_depts))
		department = pick(GLOB.available_depts)
```
Just a lot more compact, also add the text here instead of the end of the proc cause it already returns